src/test/cdk/explorer/commands/visualizeStateMachineCDK.test.ts

```diff
@@ -168,56 +174,191 @@ describe('StepFunctions VisualizeStateMachine', async function () {
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().size, 1)
+
+        //change workspace
+        mockAslVisualizationCDKManager.setWorkspaceName("Workspace2")
+        // visualization for mockStateMachineNode2
+        await mockAslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage, mockStateMachineNode)
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().size, 2)
     })
 
-    it('Test AslVisualizationCDKManager managedVisualizations set adds second Vis on different preview', async function () {
-        assert.strictEqual(aslVisualizationCDKManager.getManagedVisualizations().size, 0)
-        
-        //render graph
-        aslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage,mockStateMachineNode)
-        assert.strictEqual(aslVisualizationCDKManager.getManagedVisualizations().size, 1)
-        aslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage,mockStateMachineNode2)
-        assert.strictEqual(aslVisualizationCDKManager.getManagedVisualizations().size, 2)
+    it('Test AslVisualizationCDKManager managedVisualizationsCDK set does not add duplicate renders when multiple VisCDK active', async function () {
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().size, 0)
+
+        // visualization for mockStateMachineNode
+        await mockAslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage, mockStateMachineNode)
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().size, 1)
+
+        // visualization for mockStateMachineNode2
+        await mockAslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage, mockStateMachineNodeSameAppDiffName)
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().get(mockAslVisualizationCDKManager.getWorkspaceName())?.size, 2)
+
+        // visualization for mockStateMachineNode again
+        await mockAslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage, mockStateMachineNode)
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().get(mockAslVisualizationCDKManager.getWorkspaceName())?.size, 2)
+
+        // visualization for mockStateMachineNode2 again
+        await mockAslVisualizationCDKManager.visualizeStateMachine(mockGlobalStorage, mockStateMachineNodeSameAppDiffName)
+        assert.strictEqual(mockAslVisualizationCDKManager.getManagedVisualizations().get(mockAslVisualizationCDKManager.getWorkspaceName())?.size, 2)
     })
 
-    it('Test AslVisualizationCDKManager managedVisualizations set does not add duplicate renders when multiple Vis active', async function () {
+    it('Test AslVisualizationCDKManager managedVisualizationsCDK set removes visualization on removal of template.json file, single visCDK', async function () {
```

Comment:
Not sure if I need to (if yes, how to) test the last three cases. Implementation is left blank.